api-docs: add wallet http auction endpoints

pinheadmz opened the pull request:
Split off from https://github.com/handshake-org/handshake-org.github.io/pull/41

boymanjor commented on src/api-docs/source/includes/_wallet_auctions.md:
```diff
@@ -1,179 +1,912 @@
+
+Parameters | Description
+---------- | -----------
+id <br> _string_ | id of wallet you would like to retrieve the name list for
```
suggestion: these wallet descriptions are similar to the name descriptions. We could simplify them to "wallet id."

boymanjor commented on src/api-docs/source/includes/_wallet_auctions.md:
````diff
@@ -1,179 +1,912 @@
+]
+```
+
+List all bids for all names known to the wallet.
````
s/bids/reveals/

boymanjor commented on src/api-docs/source/includes/_wallet_auctions.md:
````diff
@@ -1,179 +1,912 @@
 ```
 
-Your wallet tracks any name on which you have bid or opened. `getnames` returns info on each. This is different from the `hsd-rpc getnames` call which returns info on all names for which the node has data.
+List the bids for a specific name known to the wallet.
````
s/bids/reveals/

boymanjor commented on src/api-docs/source/includes/_wallet_auctions.md:
````diff
@@ -1,179 +1,912 @@
+}
+```
+
+List all bids for all names known to the wallet.
````
This description is wrong.

boymanjor commented on src/api-docs/source/includes/_wallet_auctions.md:
```diff
@@ -1,179 +1,912 @@
+
+### HTTP Request
+
+`GET /wallet/:id/reveal`
```
This endpoint and following description is wrong too.

boymanjor commented on src/api-docs/source/includes/_wallet_auctions.md:
```diff
@@ -329,407 +1033,409 @@ const client = new NodeClient(clientOptions);
-name | Required | The name for which you wish to reveal your bid
+Create, sign, and send a name REVEAL. If multiple bids were placed on a name,
+all bids will be revealed by this transaction. If no value is passed in for `name`,
+all bids for all names in the wallet will be sent.
```
s/bids/reveals/

pinheadmz commented:
@boymanjor addressed all nits and did a big clean up of the wallet rpc stuff, simplifying all the descriptions and parameters, fixing lots of mistakes in the examples. Also rebased this PR so it doesnt conflict with the other PRs.
